src/bio.h

```diff
@@ -10,6 +10,26 @@
 #define __BIO_H
 
 typedef void lazy_free_fn(void *args[]);
+typedef void comp_fn(uint64_t user_data);
+
+typedef enum bio_worker_t {
+    BIO_WORKER_CLOSE_FILE = 0,
+    BIO_WORKER_AOF_FSYNC,
+    BIO_WORKER_LAZY_FREE,
+    BIO_WORKER_NUM
+} bio_worker_t;
+
+/* Background job opcodes */
+typedef enum bio_job_type_t {
+    BIO_CLOSE_FILE = 0,     /* Deferred close(2) syscall. */
+    BIO_AOF_FSYNC,          /* Deferred AOF fsync. */
+    BIO_LAZY_FREE,          /* Deferred objects freeing. */
+    BIO_CLOSE_AOF,
+    BIO_COMP_RQ_CLOSE_FILE,  /* Job completion request, registered on close-file worker's queue */
+    BIO_COMP_RQ_AOF_FSYNC,  /* Job completion request, registered on aof-fsync worker's queue */
+    BIO_COMP_RQ_LAZY_FREE,  /* Job completion request, registered on lazy-free worker's queue */
```

Comment:
@moticless i'm wondering if we need these enums which are always one-by-one with other job_type.
is the following approach reasonable?
this way we don't need to add the corresponding complete job type when we add a new job_type.
```c
void bioCreateCompRq(int job_type, comp_fn *func, uint64_t user_data);

void bioSubmitJobInternal(int type, unsigned long worker, bio_job *job) {
    job->header.type = type;
    pthread_mutex_lock(&bio_mutex[worker]);
    listAddNodeTail(bio_jobs[worker],job);
    bio_jobs_counter[type]++;
    pthread_cond_signal(&bio_newjob_cond[worker]);
    pthread_mutex_unlock(&bio_mutex[worker]);
}

void bioCreateCompRq(int type, comp_fn *func, uint64_t user_data) {
    bio_job *job = zmalloc(sizeof(*job));
    job->comp_rq.fn = func;
    job->comp_rq.arg = user_data;
    bioSubmitJobInternal(BIO_COMP_RQ, bio_job_to_worker[type], job);
}
```

Reply:
Suggested API of `bioCreateCompRq(type, ....)` assumes that any type can create completion-request which is not the case.